osconfig_tests/compute/instance.go

```diff
@@ -82,6 +88,52 @@ func (i *Instance) WaitForSerialOutput(match string, port int64, interval, timeo
 	}
 }
 
+// StreamSerialOutput stores the serial output of an instance to GCS bucket
+func (i *Instance) StreamSerialOutput(ctx context.Context, storageClient *storage.Client, logsPath, bucket string, wg *sync.WaitGroup, port int64, interval time.Duration) {
+	defer wg.Done()
+
+	logsObj := path.Join(logsPath, fmt.Sprintf("%s-serial-port%d.log", i.Name, port))
+	logger.Infof("Streaming instance %q serial port %d output to https: //storage.cloud.google.com/%s/%s", i.Name, port, bucket, logsObj)
+	var start int64
+	var buf bytes.Buffer
+	var gcsErr bool
+	tick := time.Tick(interval)
+
+Loop:
+	for {
+		select {
+		case <-tick:
+			resp, err := i.client.GetSerialPortOutput(path.Base(i.Project), path.Base(i.Zone), i.Name, port, start)
+			if err != nil {
+				// Instance is stopped or stopping.
+				status, sErr := i.client.InstanceStatus(path.Base(i.Project), path.Base(i.Zone), i.Name)
+				switch status {
+				case "TERMINATED", "STOPPED", "STOPPING":
+					if sErr == nil {
+						break Loop
+					}
+				}
+				logger.Errorf("Instance %q: error getting serial port: %v", i.Name, err)
+				break Loop
```

Comment:
I usually consider this a a really good indicator that the code should be refactored, often by factoring out a function. However, in this case, could you change the logic to something like this and remove the label?
```go
if !isTerminal(status) {
  logger.Errorf(...)
}
continue;
```

Reply:
removed the label.